### PR TITLE
53+54+65 : Test Driven Development

### PR DESCRIPTION
**Pull request by Maryam313**
Plz review the code..

### CODE REVIEW

File: src/test/java/TddWorldTest.java [hunk 1]
```diff
@@ -127,17 +121,33 @@ public void testIntPower() {
 		assertEquals(8, tdd.calculatePower(2, 3));
 	}
 	
+	@Test
```
Comment (Najlaalrajhia98) [inline src/test/java/TddWorldTest.java:122]: Well done 👍

**Review by Najlaalrajhia98 — APPROVED** [review]

File: src/test/java/TddWorldTest.java [hunk 2]
```diff
@@ -88,19 +87,14 @@ public void testOdd() {
 	@Test
 	public void testNull() {
 		TddWorld tdd = new TddWorld();
-		int[] input = {};
-		HashMap<String, ArrayList<Integer>> expectedOutput = tdd.evenOrOdd(input);
-		assertFalse(expectedOutput.isEmpty());
-	}
+		int[] input = null;
+		//HashMap<String, ArrayList<Integer>> expectedOutput = tdd.evenOrOdd(input);
```
Comment (CodelineAtyab) [inline src/test/java/TddWorldTest.java]: WE can remove this comment

File: src/test/java/TddWorldTest.java [hunk 3]
```diff
@@ -88,19 +87,14 @@ public void testOdd() {
 	@Test
```
Comment (CodelineAtyab) [inline src/test/java/TddWorldTest.java:87]: We can also make use of DisplayName("Describe that I am doing this") to make it more readable for devs

File: src/test/java/TddWorldTest.java [hunk 4]
```diff
@@ -127,17 +121,33 @@ public void testIntPower() {
...
+	
 	
 	@Test
 	@DisplayName("Test Reverse String.. ")
-	public void testReverse() {
+	public void testReverse1() {
 		TddWorld tdd = new TddWorld();
 		assertEquals("olleh", tdd.reverse("hello"));
+	}
+	@Test
```
Comment (CodelineAtyab) [inline src/test/java/TddWorldTest.java:139]: We can try to be consistent and use the same description approach as used at line 135

**Review by CodelineAtyab — CHANGES_REQUESTED** [review]

**Review by BaraahAlKalbani — APPROVED** [review]
Great Work 👍🏻

File: src/test/java/TddWorldTest.java (hunk 2 above)
Comment (Maryam313) [reply src/test/java/TddWorldTest.java]: Done!

File: src/test/java/TddWorldTest.java (hunk 3 above)
Comment (Maryam313) [reply src/test/java/TddWorldTest.java:87]: Done!

**Review by AnwaarAlbusaidi — APPROVED** [review]
great job

File: src/test/java/TddWorldTest.java (hunk 4 above)
Comment (Maryam313) [reply src/test/java/TddWorldTest.java:139]: Alright!

**Review by Najlaalrajhia98 — APPROVED** [review]

**Pull request merged**